model/src/main/proto/exploration_checkpoint.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package model;
+
+import "exploration.proto";
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Top-level proto used to store exploration checkpoint per-profile basis.
+message ExplorationCheckpointDatabase {
+
+  // map from explorationId to ExplorationCheckpoint.
+  map<string, ExplorationCheckpoint> exploration_checkpoint = 1;
+}
+
+// Corresponds to the exploration checkpoint which will be used to resume the exploration.
+// This model contains information about all states of the deck which have been visited by the user.
+message ExplorationCheckpoint {
+
+  // The name of the latest state in the deck.
+  string pending_state_name = 1;
+
+  // The list of all states which the learner has previously completed.
+  repeated CompletedStateInCheckpoint completed_states_in_checkpoint = 2;
+
+  // Stores answers to the latest state in the deck.
+  repeated AnswerAndResponse pending_user_answers = 3;
+
+  // The Index of the current selected state from the deck of states.
+  int32 state_index = 4;
+
+  // Stores -1 if no hint for the pending top state has been revealed otherwise it stores the index
+  // of the hint that the learner has revealed.
+  int32 hint_index = 5;
+
+  // Stores true if the solution to the pending top state has been revealed by the learner otherwise
+  // it stores false.
+  bool solution_is_revealed = 6;
+
+  // The timestamp in milliseconds of when the checkpoint was saved for the first time.
+  int64 timestamp_of_first_checkpoint = 7;
+
+  // Stores the title of the saved exploration.
+  string exploration_title = 8;
```

Comment:
Why do we need this title? I was thinking we can get the required information from `exploration_id` present in map.

Reply:
The exploration title will be displayed to the user when the storage runs out. If the user deletes a topic with saved progress, we won't get the exploration title.

Reply:
Okay. Thanks.